spatialmath/dualquaternion.go

```diff
@@ -185,6 +186,39 @@ func (q *dualQuaternion) Transformation(by dualquat.Number) dualquat.Number {
 	}
 }
 
+// Format implements fmt.Formatter to allow for finer grain control over printing.
+// dualquat.Number also has this implemented so in order to get custom printing we need to also implement this as opposed to simply
+// implementing fmt.Stringer.
+// Some example verbs and how they would display a zero pose are shown below
+//
+//	Verb    | String
+//	%v		| {X:0 Y:0 Z:0 OX:0 OY:0 OZ:1 Theta:0°}
+//	%.3v	| {X:0.000 Y:0.000 Z:0.000 OX:0.000 OY:0.000 OZ:1.000 Theta:0.000°}
+//	%#v		| r3.Vector{0 0 0} *spatialmath.OrientationVectorDegrees{0 0 0 1}
+//	%g		| ((1+0i+0j+0k)+(+0+0i+0j+0k)ϵ)
```

Comment:
G isn't actually an option, no? That's default.
Considering how difficult actually reading a dual quaternion is, I'm not sure it should be default...

Reply:
So the reason I had to do it this way (using `fmt.Formatter` instead of `fmt.Stringer` is because the dualquat package implemented `fmt.Formatter` (which is preferred to stringer) and has a lot of code dedicating to formating this type. I didn't want to completely get rid of that work so I made all unsupported cases in my function (i.e. g) call the dualquat package's formatting function.  If you do `fmt.Print(quaternion)` it implicitly uses `%v` so the true default case still calls our code but with this version you could still get the quaternion output if you really wanted with `fmt.Printf("%g", quaternion)`.